### ISSUE TITLE
Unable to log in using "Continue with Google"

### ISSUE THREAD

**Issue by wpshellbelle** [wordpress-mobile/WordPress-Android#14819]
### Expected behavior

I expected to click *Continue with Google* and to be able to log in using my Google account after selecting my account in the Google accounts pop-up.

### Actual behavior

Clicking *Continue with Google* appears to not be working - not able to log into Jetpack using Google. 

Video of issue:

https://user-images.githubusercontent.com/77413972/121588159-722cc900-c9ea-11eb-8b64-0ba491060857.mp4

### Steps to reproduce the behavior

1. While in the app, click *Log in with WordPress.com*
2. Select *Continue with Google* 
3. Select one of your Google accounts
4. Find that you are not logged in - still prompted to log in in app. Selecting Google account appears to not do anything.

**Comment by leandroalonso** [comment]
I can reproduce the same issue consistently (tried with different accounts and the same thing happened).

~When I try the same flow in WP it gives me a "There was some trouble connecting with the Google account.", I'm not sure if there's an issue there too.~ (this is what happens when you forget to disable your proxy settings)

@zwarm @ParaskP7 @ashiagr some of you could please take a look at this asap? Thanks!

**Comment by zwarm** [comment]
@leandroalonso - I took a very quick look to determine which was the last working release. It appears that this was working in release 17.2 (Release asset created on May 3) and not working in 17.3 (Release asset created on May 17). I don't yet know what happened in-between the two releases that could have caused this. Github tells me that the last time Google classes were updated was in late 2020 (predates the issue). We we did do a major merge of WC/WP login library changes and pushed those to the login library. That was on May 24th, which would have been after the 17.3 release, so I don't think that had any affect (but would love for someone else to take a look).

**Comment by zwarm** [comment]
@ashiagr -  a thought, do you think this could be an api/key issue with google-services.json? Let's sync up. It's probably not a google-services.json thing, but trying to capture all bases. Or maybe it is ... my local error is INVALID TOKEN. iirc, weren't we getting that for Jetpack back when we broke WP?

**Comment by leandroalonso** [comment]
@zwarm just to clarify: there are no issues with WordPress, I edited my comment — I had some wrong proxy settings enabled. Sorry about that. :|

As for Jetpack, I was able to consistently reproduce this issue. After I deleted the app and reinstalled it, it worked just fine. 🤔

@wpshellbelle could you please test again using [this version](https://play.google.com/store/apps/details?id=com.jetpack.android)?

**Comment by zwarm** [comment]
I can't log in using Google with **WordPress**. I downloaded version 17.2 and it worked fine, but it stopped working with 17.3. Latest version from Play Store works fine though.

> As for Jetpack, I was able to consistently reproduce this issue. After I deleted the app and reinstalled it, it worked just fine. 🤔

That's interesting - @leandroalonso  so an upgrade doesn't work, but a fresh install does? I will take a look at that

**Comment by ashiagr** [comment]
> I can't log in using Google with WordPress. I downloaded version 17.2 and it worked fine, but it stopped working with 17.3. Latest version from Play Store works fine though.

This is because we had a Google sign-in issue in the WordPress app in `17.3` which got fixed in `17.3.1` (Internal ref: p5T066-2hM-p2#comment-8296).

For the Jetpack app, Google login is working in [jp-0.2](https://github.com/wordpress-mobile/WordPress-Android/releases/tag/jp-0.2) created on May 21, the one which we shared here: pbMoDN-2xt-p2.

I'll share my findings for [jp-03](https://github.com/wordpress-mobile/WordPress-Android/releases/tag/jp-0.3) which was created on Jun 2.

**Comment by zwarm** [comment]
> This is because we had a Google sign-in issue in the WordPress app in `17.3` which got fixed in `17.3.1` (Internal ref: p5T066-2hM-p2#comment-8296).

OMG, I completely forgot about that hotfix - thank you for reminding me.


> For the Jetpack app, Google login is working in [jp-0.2](https://github.com/wordpress-mobile/WordPress-Android/releases/tag/jp-0.2) created on May 21, the one which we shared here: pbMoDN-2xt-p2.
👍 
 
> I'll share my findings for [jp-03](https://github.com/wordpress-mobile/WordPress-Android/releases/tag/jp-0.3) which was created on Jun 2.
🙇

**Comment by leandroalonso** [comment]
Tested today again and it's not working 🙃

Here's the log:

```
77 - [Jun-17 11:47 NUX] Google API client connected
78 - [Jun-17 11:47 NUX] LoginEmailFragment: Google API client connected
79 - [Jun-17 11:47 STATS] :large_blue_circle: Tracked: login_social_button_click
80 - [Jun-17 11:47 STATS] :large_blue_circle: Tracked: unified_login_interaction, Properties: {"source":"default","flow":"wordpress_com","step":"start","click":"login_with_google"}
81 - [Jun-17 11:47 MAIN] GOOGLE LOGIN: startFlow
82 - [Jun-17 11:47 STATS] :large_blue_circle: Tracked: unified_login_step, Properties: {"source":"default","flow":"google_login","step":"start"}
83 - [Jun-17 11:47 MAIN] LoginActivity: onActivity Result - requestCode1605477802
84 - [Jun-17 11:47 MAIN] GOOGLE LOGIN: Google has returned a sign in result - canceled
85 - [Jun-17 11:47 STATS] :large_blue_circle: Tracked: login_social_button_failure
86 - [Jun-17 11:47 NUX] Google Login Failed: result was CANCELED.
87 - [Jun-17 11:47 MAIN] GOOGLE SIGNUP/LOGIN: finishing signup/login
88 - [Jun-17 11:47 MAIN] GOOGLE SIGNUP/LOGIN: disconnecting google client
89 - [Jun-17 11:47 STATS] :large_blue_circle: Tracked: unified_login_interaction, Properties: {"source":"default","flow":"wordpress_com","step":"start","click":"show_help"}
90 - [Jun-17 11:47 STATS] :large_blue_circle: Tracked: unified_login_step, Properties: {"source":"default","flow":"wordpress_com","step":"help"}
91 - [Jun-17 11:47 STATS] :large_blue_circle: Tracked: support_opened, Properties: {"origin":"LOGIN_SOCIAL"}
```

**Comment by ashiagr** [comment]
Findings added to internal ref: p1623870755164700-slack-C0180B5PRJ4

**Comment by ashiagr** [comment]
This is resolved and doesn't need a new version upload, more details in thread p1623870755164700-slack-C0180B5PRJ4.
Cc @AliSoftware, special mention to @jkmassel 🙇‍♀️.

@wpshellbelle, @thehenrybyrd, can one of you please test [the available version on Google Play](https://play.google.com/store/apps/details?id=com.jetpack.android) so we can close this issue?

**Comment by AliSoftware** [comment]
This should now be fixed in 17.5 and in 17.6-rc-2

**Comment by leandroalonso** [comment]
> This should now be fixed in 17.5 and in 17.6-rc-2

Tested and it's working.

**Comment by thehenrybyrd** [comment]
Working for me too in 17.5 :)
